helm/charts/hpe-csi-driver/values.yaml

```diff
@@ -33,4 +33,4 @@ kubeletRootDir: "/var/lib/kubelet/"
 disableNodeGetVolumeStats: false
 
 # CSP client timeout for HPE Alletra 9000, Primera and 3PAR (60-360 seconds)
-cspClientTimeout: 60
+cspClientTimeout: 360
```

Comment:
Why should this parameter even be configurable? What would the reason be to dial down the timeout?

Reply:
Good point, it could have been because a developer's mind thinks about making just about everything configurable to avoid any situation where different situations require changes to the parameter.

This parameter has been there since last release. Should we be removing this now ?

Reply:
@raunakkumar what is your take here? The downside is that errors will take 6 minutes to croak by default if something tarpits but having this configurable doesn't make any sense either.

Reply:
It was added earlier for Alletra 9000 when the workflows were timing out (the comment mentions 3par/Primera). We still wanted to keep the default 60 for Alletra 6000/Nimble.

Reply:
> 

@datamattsson  - i think its one kind of error that would take 6 minutes; which is if the csp is busy or its dead. If that was a functional/business logic error from CSP, it would return immediately.  In the situation of csp being busy, the high timeout is actually serving purpose. The only problem comes if the CSP dies (and discards response to this request), so i think that situation is very rare, and we should be ok with the higher timeout.

If you want @maddiegos will remove the configurability and make it always 360 seconds. Let him know.

Reply:
I'm fine removing it and setting whatever sane default make sense. Where does 360 seconds come from? How will Kubernetes react when waiting 6 minutes for an operation to return?

Please make sure to remove it from readme and schema too.

Reply:
Good we will do that..

Reply:
I am little concerned making it 360 seconds for Nimble CSP. we have never seen an instance of csp response > few seconds and in many scenarios kubernetes have timeout of ~ 2 mins. If we do change this to 360 we need to make sure CSP workflows have to be idempotent.

Reply:
I think we have arrived at the point where need to split the charts.

Reply:
@raunakkumar - This, as you have already pointed out, is not applicable to nimble csp. So we should be good.

Reply:
> Well, leave the value where it is and we have to educate 3PAR customers to tune this timeout.

Agreed

Reply:
@raunakkumar this [blatant hack](https://github.com/hpe-storage/csi-driver/blob/ae0812a550cd3e569fef38444e4818e28a624473/pkg/driver/driver.go#L361) makes this parameter null and void any `Service` that contains "alletra9000" or "primera". So, are we back to removing the parameter from the chart?

Reply:
Thanks for bringing this up . Yes, we can do without the parameters. It needs definitive testing if we have both CSP enabled on same cluster.

Reply:
The "blatant hack" was required to keep the timeout change only to the 3PAR/Primera CSP. Understanding that we dont like this, @datamattsson - can you suggest a better approach to accomplish the same ?
Note: default timeout of 60 mins is out of question as it does not work for us.

Reply:
@datamattsson  - Calling it a "blatant hack" has negative connotations attached to it :) Do you have a better solution you propose to accomplish the same ?

Reply:
modified